### PR TITLE
feat(api): Use deck-absolute coords in hardware_control

### LINKED ISSUE #2238: Robot Singleton Refactor: hardware_control should take in deck-absolute coordinates
The hardware_control module should track the deck calibration and take deck-absolute coordinates as inputs rather than symbolic names of locations. It should be responsible only for figuring out how to move a pipette critical point to a deck-absolute location.

### PR DESCRIPTION
**Pull request by sfoster1**
## Overview

This commit adds the deck calibration and mount offset to the hardware controller. `opentrons.hardware_control.API.move_to` and `opentrons.hardware_control.API.move_rel` now use these transforms.

This has a couple follow-on effects:

- `robot_config.py` is moved to `opentrons.config` since it's now shared between the legacy api and the new api (and this way the configurations are more centralized)
- We can actually move the robot with `hardware_control` now! You still need to connect to the smoothie manually by creating an api and then calling `._backend._smoothie_driver.connect()` before you can move (this is addressed in another PR) but once you do, you can home and move around.

## Review Requests

Put this on a robot, kill the api server, and try moving it around:
```
# python
>>> import opentrons.hardware_control as hc
>>> import opentrons.types as t
>>> import asyncio
>>> l = asyncio.get_event_loop()
>>> a = hc.API.build_hardware_controller()
>>> a._backend._smoothie_driver.connect()
>>> l.run_until_complete(a.home())
>>> l.run_until_complete(a.move_to(t.Mount.RIGHT, t.Point(12.13, 9, 150)) # the front left cal point
>>> l.run_until_complete(a.move_to(t.Mount.LEFT, t.Point(12.13, 9, 150))
```

Both the left and right moves should leave the appropriate pipette in the appropriate place. Right now, this is just moving mounts, so there's no consideration for single vs multi pipettes - the center of the mount should be over the calibration point (at least if the machine has deck calibration).

In addition, you should be able to move relative:

```
>>> l.run_until_complete(a.move_rel(t.Mount.RIGHT, t.Point(40, 0, 0))) # move 40mm right
```

And get each mount's location, noting that the mount offset is properly applied (numbers may vary based on your robot's mount offset):

```
>>> a.current_position(t.Mount.RIGHT)
{<Axis.X: 1>: 86.13, <Axis.Y: 2>: 9, <Axis.A: 4>: 100}
>>> a.current_position(t.Mount.LEFT)
{<Axis.X: 1>: 52.129999999999995, <Axis.Y: 2>: 9, <Axis.Z: 3>: 150}
```

Closes #2238

### CODE REVIEW

**Comment by sanni-t** [comment]
Commands worked successfully on sanni-t bot  👍
Not entirely clear why `ps | grep opentrons.main | xargs kill` is needed

**Comment by sfoster1** [comment]
> Commands worked successfully on sanni-t bot 👍
> Not entirely clear why `ps | grep opentrons.main | xargs kill` is needed

They're needed because the main server is holding the smoothie connection open and you need to kill it so you can connect instead.

File: api/src/opentrons/hardware_control/__init__.py [hunk 1]
```diff
@@ -183,52 +177,199 @@ def get_connected_hardware(self):
...
+        """
+        with_enum = {Axis[k]: v for k, v in smoothie_pos.items()}
+        other = {k: v for k, v in with_enum.items()
+                 if k not in Axis.gantry_axes()}
+        right = (with_enum[Axis.X], with_enum[Axis.Y],
+                 with_enum[Axis.by_mount(top_types.Mount.RIGHT)])
+        # Tell apply_transform to just do the change of base part of the
+        # transform rather than the full affine transform, because this is
+        # an offset
+        offset_in_smoothie = linal.apply_transform(
```
Comment (btmorr) [inline api/src/opentrons/hardware_control/__init__.py]: (summarizing f2f convo) I think the mount offset should be handled in deck coordinates for now. We will possibly/probably have to take z-tilt into consideration at some point, which might make mount offset need to be calculated in smoothie coordinates, but that isn't clear yet. Also, when the left pipette and the right pipette go to the same spot, the gantry ends up in a different position for each, so z-tilt would have to be calculated at a different point. Altogether, I think we should wait until we have a better understanding of what we're going to have to do about z tilt before incorporating this.

File: api/src/opentrons/hardware_control/__init__.py [hunk 2]
```diff
@@ -183,52 +177,199 @@ def get_connected_hardware(self):
...
+        mount would depend on its actual position in deck frame, so we have
+        to apply the mount offset.
+
+        TODO: Figure out which frame the mount offset is measured in, because
+              if it's measured in the deck frame (e.g. by touching off points
+              on the deck) it has to go through the reverse transform to be
+              added to the smoothie coordinates here.
+        """
+        with_enum = {Axis[k]: v for k, v in smoothie_pos.items()}
+        other = {k: v for k, v in with_enum.items()
```
Comment (btmorr) [inline api/src/opentrons/hardware_control/__init__.py]: Mind calling this "pluger_axes" or something?

**Comment by btmorr** [comment]
🐺

**Pull request merged**